charms/worker/k8s/src/token_distributor.py

```diff
@@ -275,24 +308,30 @@ def revoke_tokens(
             token_strategy (TokenStrategy): The strategy of token creation.
             token_type (ClusterTokenType, optional): The type of cluster token.
                 Defaults to ClusterTokenType.NONE.
-            to_remove (Set[ops.Unit], optional): Specific units to ensure its token is revoked
+            to_remove (ops.Unit, optional): unit to ensure its token is revoked
 
         Raises:
             ValueError: If an invalid token_strategy is provided.
         """
+        # any unit currently in the relation
+        all_units = relation.units
+        if self.charm.app == relation.app:
+            # include self in peer relations
+            all_units |= {self.charm.unit}
+
+        # any unit in the app_databag, which successfully recovered its token
         app_databag = relation.data[self.charm.app]
         joined = {self.charm.model.get_unit(str(u)) for u in app_databag}
-        if to_remove is not None:
-            remaining = joined - to_remove
-        else:
-            remaining = joined & (relation.units | {self.charm.unit})
-        remove = joined - remaining
+
+        # establish the remaining units
+        remove = {to_remove} if to_remove else (joined - all_units)
```

Comment:
if i'm told to remove one unit -- only try to remove that ONE unit -- ignore trying to reap all the others for now